### PR TITLE
WIP packaging for Agimus

### PR DESCRIPTION
**Pull request by longhathuc**
Added packages of agimus to wip, built on ferrum's environment
- agimus
- agimus-hpp
- agimus-sot
- agimus-vision
- agimus-demos
- agimus_sot_msgs

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -21,28 +21,25 @@ SUBDIR+=	hatponboard-lib
...
-SUBDIR+=	hpp-benchmark
-SUBDIR+=	hrp2-14-description
-SUBDIR+=	hrp2-dev
-SUBDIR+=	infuse-asn1-conversions
-SUBDIR+=	infuse-asn1-types
-SUBDIR+=	infuse-envire
-SUBDIR+=	infuse-idl
-SUBDIR+=	infuse-msgs-ros
-SUBDIR+=	infuse-novatel-gps-driver-ros
+#SUBDIR+=	infuse-asn1-conversions
```
Comment (nim65s) [inline Makefile]: Please don't comment the packages you are not using: they will be used on the buildfarm

File: agimus-demos/Makefile [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# robotpkg Makefile for:	wip/agimus-vision
+# Created:			Long Ha Thuc on Tue, 18 Aug 2020
+#
+
+ORG=			agimus
+NAME=			agimus-demos
+VERSION=		1.1.0
+DOWNLOAD_LOC=   releases/download
+MASTER_SITE_AGIMUS=     https://github.com/agimus/
```
Comment (nim65s) [inline agimus-demos/Makefile]: No need for this: we already have `MASTER_SITE_GITHUB`

File: agimus-demos/Makefile [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# robotpkg Makefile for:	wip/agimus-vision
+# Created:			Long Ha Thuc on Tue, 18 Aug 2020
+#
+
+ORG=			agimus
+NAME=			agimus-demos
+VERSION=		1.1.0
+DOWNLOAD_LOC=   releases/download
```
Comment (nim65s) [inline agimus-demos/Makefile]: No need for this: we usually put:
```
MASTER_SITES?=    ${MASTER_SITE_GITHUB:=${ORG}/${NAME}}/releases/download/v${VERSION}/
```

File: agimus-demos/Makefile [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+MASTER_SITE_AGIMUS=     https://github.com/agimus/
+DISTNAME=		${NAME}-${VERSION}
+
+MASTER_SITES=		${MASTER_SITE_AGIMUS:=${NAME}/${DOWNLOAD_LOC}/v${VERSION}/}
+MASTER_REPOSITORY=	${MASTER_REPOSITORY_GITHUB}${ORG}/${NAME}/${DOWNLOAD_LOC}/v${VERSION}
```
Comment (nim65s) [inline agimus-demos/Makefile]: `MASTER_REPOSITORY` should point to a .git tree: it is used for `make checkout`. Here, you can use:
```
MASTER_REPOSITORY?=	${MASTER_REPOSITORY_GITHUB}${ORG}/${NAME}.git
```

File: agimus-demos/Makefile [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+LICENSE=		BSD-2-Clause
```
Comment (nim65s) [inline agimus-demos/Makefile]: In this field, you can only put files that are in the `licenses` directory of the main robotpkg repository. Here, you need `2-clause-bsd`

File: agimus-demos/Makefile [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+CATEGORIES=		wip
+COMMENT=		Agimus-demos
+CMAKE_ARGS+=    -DPYTHON_STANDARD_LAYOUT=ON
```
Comment (nim65s) [inline agimus-demos/Makefile]: I think I created this `PYTHON_STANDARD_LAYOUT` in the jrl-cmakemodule, but it doesn't look like agimus-sot is using that module. Why do you need this arg ?

File: agimus-hpp/Makefile [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+#
+ORG=			agimus
+NAME=			agimus-hpp
+VERSION=		1.0.0
+DOWNLOAD_LOC=   releases/download
+MASTER_SITE_AGIMUS=     https://github.com/agimus/
+DISTNAME=		${NAME}-${VERSION}
+
+MASTER_SITES=		${MASTER_SITE_AGIMUS:=${NAME}/${DOWNLOAD_LOC}/v${VERSION}/}
+MASTER_REPOSITORY=	${MASTER_REPOSITORY_GITHUB}${ORG}/${NAME}/${DOWNLOAD_LOC}/v${VERSION}
```
Comment (nim65s) [inline agimus-hpp/Makefile]: Same here

File: agimus-hpp/Makefile [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+LICENSE=		BSD-2-Clause
+
+CATEGORIES=		wip
+COMMENT=		Package to link HPP to the Agimus framework
+
+CMAKE_ARGS+=    -DPYTHON_STANDARD_LAYOUT=ON
```
Comment (nim65s) [inline agimus-hpp/Makefile]: This agimus-hpp package uses the jrl-cmakemodule. Therefore, you can use:
```
include ../../devel/jrl-cmakemodules/Makefile.common
```
to setup everything you need: this `-DPYTHON_STANDARD_LAYOUT=ON` is automatically added there, among other features

File: agimus-sot/depend.mk [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+PREFER.agimus-sot?=		robotpkg
+
+DEPEND_USE+=			agimus-sot
+
+DEPEND_ABI.agimus-sot?=	agimus-sot>=1.0.2
+DEPEND_DIR.agimus-sot?=	../../wip/agimus-sot
+
+SYSTEM_SEARCH.agimus-sot=	\
+	'lib/pkgconfig/agimus-sot.pc:/Version/s/[^0-9.]//gp' \ 
+	'lib/lib/libagimus_sot.so'
```
Comment (nim65s) [inline agimus-sot/depend.mk]: `lib/lib/` ? This looks like a typo

File: agimus-vision/Makefile [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+
+ORG=			agimus
+NAME=			agimus-vision
+VERSION=		1.0.0
+DOWNLOAD_LOC=   releases/download
+MASTER_SITE_AGIMUS=     https://github.com/agimus/
+DISTNAME=		${NAME}-${VERSION}
+
+MASTER_SITES=		${MASTER_SITE_AGIMUS:=${NAME}/${DOWNLOAD_LOC}/v${VERSION}/}
+MASTER_REPOSITORY=	${MASTER_REPOSITORY_GITHUB}${ORG}/${NAME}/${DOWNLOAD_LOC}/v${VERSION}
```
Comment (nim65s) [inline agimus-vision/Makefile]: same here

File: sysdep/pal-distro.mk [hunk 11]
```diff
@@ -6,8 +6,8 @@ ifndef _paldistro_mk
   _paldistro:=$(shell ${SHELL} ${ROBOTPKG_DIR}/wip/sysdep/pal-distro.sh)
   export PALDISTRO:=  $(word 1,${_paldistro})
 
-  ifeq (erbium,${PREFER.pal-distro})	
-     ifeq (palerbium,${PALDISTRO})
+  ifeq (ferrum,${PREFER.pal-distro})	
+     ifeq (palferrum,${PALDISTRO})
```
Comment (nim65s) [inline sysdep/pal-distro.mk:10]: I don't think you should remove anything for erbium

**Review by nim65s — CHANGES_REQUESTED** [review]

**Comment by longhathuc** [comment]
I have modified the files as your suggestions

**Comment by longhathuc** [comment]
on robotpkg.conf, I have added these lines

PREFER.py-prf-ros-controllers = robotpkg
PREFER.py-prf-ros-control = robotpkg
PREFER.ros-four-wheel-steering-msgs = robotpkg
PREFER.ros-urdf-geometry-parser = robotpkg
PREFER.ros-actionlib = robotpkg
PREFER.ros-class-loader = robotpkg
PREFER.ros-cmake-modules = robotpkg
PREFER.ros-comm = robotpkg
PREFER.ros-common-msgs = robotpkg
PREFER.ros-genmsg = robotpkg
PREFER.ros-genpy = robotpkg
PREFER.ros-message-generation = robotpkg
PREFER.ros-message-runtime = robotpkg
PREFER.ros-pluginlib = robotpkg
PREFER.ros-ros = robotpkg
PREFER.ros-std-msgs = robotpkg
PREFER.ros-urdf = robotpkg
PREFER.ros-roscpp-core = robotpkg
PREFER.ros-rospack = robotpkg
PREFER.prf-teleop-tools = robotpkg
PREFER.play-motion = system
PREFER.prf-teleop-tools = system
PREFER.pal-gripper = robotpkg
PREFER.ros-orocos-kinematics-dynamics = robotpkg
PREFER.ros-vision-opencv = robotpkg

File: Makefile [hunk 12]
```diff
@@ -21,9 +27,6 @@ SUBDIR+=	hatponboard-lib
 SUBDIR+=	head-action
 SUBDIR+=	hey5-description
 SUBDIR+=	hpp-affordance
-SUBDIR+=	hpp-benchmark
-SUBDIR+=	hrp2-14-description
-SUBDIR+=	hrp2-dev
```
Comment (nim65s) [inline Makefile]: This still shouldn't be removed. Nothing should be removed or commented in this file.

File: agimus_sot_msgs/Makefile [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+LICENSE=		2-clause-bsd
+
+CATEGORIES=		wip
+COMMENT=		Messages and services for the agimus_sot package
+
+include ../../devel/ros-catkin/depend.mk
+include ../../mk/language/c.mk
+include ../../mk/language/c++.mk
+include ../../mk/robotpkg.mk
+include ../../devel/jrl-cmakemodules/Makefile.common
```
Comment (nim65s) [inline agimus_sot_msgs/Makefile]: Could you put this as the first include ? I'm not sure, but I think this defines important constants for the other includes.

File: play-motion/Makefile [hunk 14]
```diff
@@ -30,6 +30,11 @@ USE_BOOST_LIBS=		thread
 ifeq (,$(filter 16.04%,${OS_VERSION}))
   include ../../wip/ros-moveit/depend.mk
 endif
+
+ifeq (,$(filter 18.04%,${OS_VERSION}))
```
Comment (nim65s) [inline play-motion/Makefile]: I don't understand this addition

**Review by nim65s — CHANGES_REQUESTED** [review]
Thanks, that's better :)
We still have a few tiny details.

**Comment by longhathuc** [comment]
I have modified the files as your suggestions

**Pull request merged**